kbfsgit/fs_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package kbfsgit
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/keybase/kbfs/libfs"
+	"github.com/keybase/kbfs/libkbfs"
+	"github.com/keybase/kbfs/tlf"
+	"github.com/stretchr/testify/require"
+	gogit "gopkg.in/src-d/go-git.v4"
+	gogitcfg "gopkg.in/src-d/go-git.v4/config"
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+	"gopkg.in/src-d/go-git.v4/storage"
+	"gopkg.in/src-d/go-git.v4/storage/filesystem"
+)
+
+func makeFS(t *testing.T, subdir string) (
+	context.Context, *libkbfs.TlfHandle, *libfs.FS) {
+	ctx := libkbfs.BackgroundContextWithCancellationDelayer()
+	config := libkbfs.MakeTestConfigOrBust(t, "user1", "user2")
+	h, err := libkbfs.ParseTlfHandle(ctx, config.KBPKI(), "user1", tlf.Private)
+	require.NoError(t, err)
+	fs, err := libfs.NewFS(ctx, config, h, subdir, "")
+	require.NoError(t, err)
+	return ctx, h, fs
+}
+
+// configInMemoryStorer keeps the git config in memory, to work around
+// a gcfg bug (used by go-git when reading configs from disk) that
+// causes a freakout when it sees backslashes in git file URLs.
```

Comment:
When reading the config from disk, gcfg fails if it sees any backslashes in any of the remote names.  When there's no config on disk, it makes a new one, so it can handle when the config is missing.

But, in a future PR that I'll put up soon, I realized we do need to write the config to disk, if only to get the "bare = true" line.  So that PR changes this class into something that just filters out the remotes before writing it to disk.  We want to do that anyway, since we don't want to start any remotes permanently in the bare repo.